components/sharing-dialog/src/autocomplete/sharing-autocomplete.test.js

```diff
@@ -0,0 +1,59 @@
+import { CustomDataProvider } from '@dhis2/app-runtime'
+import { render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import '@testing-library/jest-dom'
+import React, { useState } from 'react'
+import { SharingAutocomplete } from './sharing-autocomplete.js'
+
+jest.mock('../helpers/index.js', () => ({
+    debounce: (fn) => fn,
+}))
+
+const Wrapper = () => {
+    const [selected, setSelected] = useState()
+
+    return (
+        <SharingAutocomplete
+            selected={selected}
+            onSelection={(user) => setSelected(user?.displayName)}
+        />
+    )
+}
+
+describe('SharingAutocomplete', () => {
+    it('hides autocompletion results after selection', async () => {
+        const userDisplayName = 'Some User'
+        const dataProviderData = {
+            'sharing/search': jest.fn(() => ({
+                users: [
+                    {
+                        id: 'user-1',
+                        displayName: userDisplayName,
+                    },
+                ],
+            })),
+        }
+        render(
+            <CustomDataProvider data={dataProviderData}>
+                <Wrapper />
+            </CustomDataProvider>
+        )
+
+        const searchString = userDisplayName.slice(0, 3)
+        userEvent.type(screen.getByRole('textbox'), searchString)
+        expect(screen.getByRole('textbox')).toHaveValue(searchString)
+
+        await waitFor(() => screen.getByRole('listitem'))
+        userEvent.click(screen.getByRole('listitem').querySelector('a'))
+        expect(screen.getByRole('textbox')).toHaveValue(userDisplayName)
+
+        try {
+            await waitFor(() => screen.getByRole('listitem'), { timeout: 1 })
+        } catch (error) {
+            if (!error.message.startsWith('Unable to find role="listitem"')) {
+                throw error
+            }
+        }
```

Comment:
What we're trying to do here is wait for the two `useEffect`s in `SharingAutocomplete` to have been triggered (the 1st `useEffect` triggered is [this one](https://github.com/dhis2/ui/blob/9c4e0c428cf731ff6d18a042d3c397ce9ae63d87/components/sharing-dialog/src/autocomplete/sharing-autocomplete.js#L30), which then triggers [the second](https://github.com/dhis2/ui/blob/9c4e0c428cf731ff6d18a042d3c397ce9ae63d87/components/sharing-dialog/src/autocomplete/sharing-autocomplete.js#L39-L50)). 

An alternative way of waiting for these two `useEffect`s is to do the following:
```js
const sleep = ms => new Promise(resolve => setTimeout(resolve, ms))
await sleep(0)
```

Reply:
Yeah it's always hard to test for something not being there. So, if what we're trying to do is wait for logic that will have executed after a set amount of renders, we could also use `await waitForNextUpdate()`.

Would that be an option here?

Reply:
Unfortunately `waitForNextUpdate` is only for react hook testing library not react testing library :(

Reply:
Ah I figured they'd have something like that for react-testing as well. And how about https://testing-library.com/docs/dom-testing-library/api-async/#waitforelementtoberemoved?

Reply:
Ok I see yeah